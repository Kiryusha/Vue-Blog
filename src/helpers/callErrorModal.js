// calling error modal
export default function (context, error) {
  let errText = 'Неизвестная ошибка.';

  if (error.response) {
    errText = error.response.data.message ? error.response.data.message :
      error.response.data;
  } else {
    errText = error.message;
  }

  context.$Progress.fail();
  context.$modal.show('response', { message: errText });
}
