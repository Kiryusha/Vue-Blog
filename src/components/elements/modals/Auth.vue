<template lang="pug">
  modal(
    name="auth",
    width="100%",
    height="auto"
  )
    ModalWrapper._auth
      .content
        .title Авторизация
        .login
          Login
        .socials
          Button(
            :view="'github-auth'",
            @click="submit('github')"
          ) Я у мамы программист
          Button(
            :view="'google-auth'",
            @click="submit('google')"
          ) google
        .close
          Button(
            @click="$modal.hide('auth')"
          ) Закрыть
</template>

<script>
import { mapActions } from 'vuex';
import ModalWrapper from 'Components/elements/modals/Modal';
import Button from 'Components/elements/general/Button';
import Login from 'Components/elements/auth/Login';
import callErrorModal from '@/helpers/callErrorModal';

export default {
  components: {
    ModalWrapper,
    Button,
    Login,
  },
  data() {
    return {
      title: '',
      response: '',
    };
  },
  methods: {
    ...mapActions(['authenticate']),
    submit(provider) {
      this.authenticate({ provider }).catch((error) => {
        callErrorModal(this, error);
      });

      this.$modal.hide('auth');
    },
  },
};
</script>
