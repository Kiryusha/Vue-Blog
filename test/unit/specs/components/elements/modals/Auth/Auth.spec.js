// auth.vue

import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import Auth from 'Components/elements/modals/Auth.vue';
import callErrorModal from '@/helpers/callErrorModal';

const localVue = createLocalVue();
localVue.use(Vuex);

jest.mock('@/helpers/callErrorModal', () => jest.fn());

describe('Auth.vue', () => {
  const submit = jest.fn();
  const config = {
    localVue,
    stubs: ['modal'],
    mocks: {
      $modal: {
        hide: jest.fn(),
      },
    },
  };

  let storeConfig = {
    actions: {
      authenticate: () => {},
    },
  };
  let store = new Vuex.Store(storeConfig);

  describe('component shallow rendering', () => {
    it('should match a snapshot', () => {
      const wrapper = shallowMount(Auth, { ...config, store });

      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('custom actions', () => {
    it('should call authenticate() on submit()', () => {
      const authenticate = jest.fn().mockResolvedValue();
      const wrapper = shallowMount(Auth, { ...config, store, methods: {
        authenticate,
      }});

      wrapper.vm.submit('test')

      expect(authenticate).toHaveBeenCalledTimes(1);
    });
  });

  describe('custom actions', () => {
    it('should call callErrorModal() on submit() when get error', async () => {
      const authenticate = jest.fn().mockRejectedValue(new Error('test'));
      const wrapper = shallowMount(Auth, { ...config, store, methods: {
        authenticate,
      }});

      await wrapper.vm.submit('test')

      expect(callErrorModal).toHaveBeenCalledTimes(1);
    });
  });
});
