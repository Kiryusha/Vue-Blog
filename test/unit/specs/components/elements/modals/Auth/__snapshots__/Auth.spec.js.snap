// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth.vue component shallow rendering should match a snapshot 1`] = `<modal-stub name="auth" width="100%" height="auto"></modal-stub>`;
